apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -483,16 +483,17 @@ type CredentialsStatus struct {
 	Version   string                  `json:"credentialsVersion,omitempty"`
 }
 
-// RebootAnnotation defines known variations of reboot modes
-type RebootAnnotation string
+// RebootMode defines known variations of reboot modes
+type RebootMode string
 
 const (
-	RebootAnnotationHard RebootAnnotation = "hard"
-	RebootAnnotationSoft RebootAnnotation = "soft"
+	RebootModeHard RebootMode = "hard"
+	RebootModeSoft RebootMode = "soft"
+	RebootModeNone RebootMode = ""
```

Comment:
I don't think you need this specified explicitly, because the zero-value of a string is always an empty string in golang:

https://tour.golang.org/basics/12

Reply:
thanks @hardys I was just following precedent, e.g. https://github.com/metal3-io/baremetal-operator/blob/master/apis/metal3.io/v1alpha1/baremetalhost_types.go#L156

I'm using this comparison to return a RebootMode even when there's no reboot requested, as it helps with the tests, but I can revert if you think best.

Reply:
I've cleaned this up as you're right, it's not required even if other statuses have it.